File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = nums[0];  // Stores the maximum sum found
        int currentSum = 0;     // Tracks sum of current subarray

        for (int num : nums) {
            if (currentSum < 0) 
                currentSum = 0; // Reset if the sum is negative
            
            currentSum += num;  // Add current element
            maxSum = max(maxSum, currentSum); // Update max sum
        }

        return maxSum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxsum = nums[0]
        currsum = 0
        for n in nums:
            if currsum < 0:
                currsum = 0
            currsum += n
            maxsum = max(maxsum, currsum)

        return maxsum        
        